preprocessors/openstreetmap/osm_service.py

```diff
@@ -314,79 +334,93 @@ def get_amenities(bbox_coord):
                 logging.error(error)
                 amenities = None
 
-    # Filter the amenity tags to the basic useful ones
+    # Fetch the basic amenity tags
     amenity = []
     if amenities is not None:
         if amenities.nodes:
             for node in amenities.nodes:
-                if node.tags.get("amenity") is not None:
-                    amenity_record = {
-                        "id": int(node.id),
-                        "lat": float(node.lat),
-                        "lon": float(node.lon),
-                        "name": node.tags.get("name"),
-                        "cat": node.tags.get("amenity"),
-                    }
-                    # Fetch as many tags possible
-
-                    for key, value in node.tags.items():
-                        if value != node.tags.get(
-                                "name") and value != node.tags.get("amenity"):
-                            if key not in amenity_record:
-                                amenity_record[key] = value
-
-                # Delete keys with no value
-                amenity_record = dict(
-                    x for x in amenity_record.items() if all(x))
-                amenity.append(amenity_record)
+                # Extract only amenities(under nodes) within the boundary
+                if node.lat >= lat_min and node.lat <= lat_max:
```

Comment:
Recommend including lat and lon checking as one `if` statement. There isn't a case where we would care about being in bounds for one and not the other and this makes indentation problems more likely during future maintenance.

Reply:
Changes effected as recommended.